palindrome/lib/palindrome.ex

```diff
@@ -28,41 +29,72 @@ defmodule Palindrome do
 
   def compute(input, :even) do
     input
-    |> reverse()
+    |> reverse(:even)
     |> check(input)
     |> case do
-      {true, new} -> to_integer(new)
+      {true, new} -> {:ok, to_integer(new)}
       {false, new} -> increment(new, :even)
     end
   end
 
-  def reverse(input) do
+  def compute(input, :odd) do
+    input
+    |> reverse(:odd)
+    |> check(input)
+    |> case do
+      {true, new} -> {:ok, to_integer(new)}
+      {false, new} -> increment(new, :odd)
+    end
+  end
+
+  def reverse(input, :even) do
```

Comment:
This is not simply `reverse-ing` the input right? You intend to convey something else.